### PR TITLE
Fix parsing of threads in stress/memory tests

### PR DESCRIPTION
**Pull request by vurusovs**
### Tickets:
 - 32295

### CODE REVIEW

File: tests/stress_tests/common/utils.cpp [hunk 1]
```diff
@@ -25,11 +25,12 @@ std::string fileNameNoExt(const std::string &filepath) {
 
 
 static size_t parseLine(char* line) {
-    // This assumes that a digit will be found and the line ends in " Kb".
     size_t i = strlen(line);
     const char* p = line;
     while (*p <'0' || *p > '9') p++;
-    line[i-3] = '\0';
+    if (line[i-1] == 'b')
```
Comment (asomsiko) [inline tests/stress_tests/common/utils.cpp:31]: it should be checking for both K and b and that the length > 2

File: tests/stress_tests/common/utils.cpp [hunk 2]
```diff
@@ -25,11 +25,12 @@ std::string fileNameNoExt(const std::string &filepath) {
 
 
 static size_t parseLine(char* line) {
-    // This assumes that a digit will be found and the line ends in " Kb".
     size_t i = strlen(line);
     const char* p = line;
     while (*p <'0' || *p > '9') p++;
```
Comment (asomsiko) [inline tests/stress_tests/common/utils.cpp:30]: could we add a check pointer p is still inside string?

**Review by asomsiko — CHANGES_REQUESTED** [review]

File: tests/stress_tests/common/utils.cpp (hunk 1 above)
Comment (vurusovs) [reply tests/stress_tests/common/utils.cpp]: Done

File: tests/stress_tests/common/utils.cpp (hunk 2 above)
Comment (vurusovs) [reply tests/stress_tests/common/utils.cpp]: Done

File: tests/stress_tests/common/utils.cpp [hunk 3]
```diff
@@ -25,13 +25,16 @@ std::string fileNameNoExt(const std::string &filepath) {
...
 static size_t parseLine(char* line) {
-    // This assumes that a digit will be found and the line ends in " Kb".
-    size_t i = strlen(line);
+    size_t len = strlen(line);
     const char* p = line;
-    while (*p <'0' || *p > '9') p++;
-    line[i-3] = '\0';
-    i = (size_t)atoi(p);
-    return i;
+    while (*p < '0' || *p > '9') p++;
```
Comment (asomsiko) [inline tests/stress_tests/common/utils.cpp:30]: we need to stop if p runs to line+ length

**Review by asomsiko — CHANGES_REQUESTED** [review]

**Review by asomsiko — APPROVED** [review]

File: tests/stress_tests/common/utils.cpp [hunk 4]
```diff
@@ -24,14 +24,19 @@ std::string fileNameNoExt(const std::string &filepath) {
...
-    line[i-3] = '\0';
-    i = (size_t)atoi(p);
-    return i;
+    while ((p < line + len) && (*p < '0' || *p > '9')) p++;
+    if (p == line + len)
+        // If number wasn't found return -1
+        return -1;
+    if (len > 2 && line[len-2] == 'K' && line[len-1] == 'b')
+        // If line ends in " Kb" remove suffix
+        line[len-3] = '\0';
```
Comment (vurusovs) [inline tests/stress_tests/common/utils.cpp:37]: check sefault

Comment (vurusovs) [reply tests/stress_tests/common/utils.cpp]: Replaced with `string` due `char*` is static and can't be modified

**Pull request merged**